examples/material_gallery/lib/demo/buttons_demo.dart

```diff
@@ -3,214 +3,267 @@
-  "it changes.";
+import '../gallery/demo.dart';
 
 const String _raisedText =
-  "A raised button is typically a rectangular material button that lifts "
-  "and displays ink reactions on press. Raised buttons add dimension to "
-  "mostly flat layouts. They emphasize functions on busy or wide spaces.";
+  "# Raised buttons\n"
+  "Raised buttons add dimension to mostly flat layouts. They emphasize"
+  "functions on busy or wide spaces.";
+
+const String _raisedCode =
+"""// Create a new flat button
+new RaisedButton(
+  child: new Text('BUTTON TITLE'),
+  onPressed: () {
+    // Perform some action
+  }
+);
+
+// Create disabled button
+new RaisedButton(
+  child: new Text('BUTTON TITLE')
+);""";
 
 const String _flatText =
-  "A flat button is made of ink that displays ink reactions on press "
-  "but does not lift. Use flat buttons on toolbars, in dialogs and "
-  "inline with padding";
+  "# Flat buttons\n"
+  "Flat buttons are printed in material. They do not lift, but fill with "
+  "color on press.\n\n"
+  "Use flat buttons in the following locations:\n\n"
+  "- On toolbars\n"
+  "- In dialogs, to unify the button action with the dialog content\n"
+  "- Inline, with padding, so the user can easily find them";
 
-const String _dropdownText =
-  "A dropdown button selects between multiple selections. The button "
-  "displays the current state and a down arrow.";
-
-class _ButtonDemo {
-  _ButtonDemo({ this.title, this.text, this.builder }) {
-    assert(title != null);
-    assert(text != null);
-    assert(builder != null);
+const String _flatCode =
+"""// Create a new flat button
+new FlatButton(
+  child: new Text('BUTTON TITLE'),
+  onPressed: () {
+    // Perform some action
   }
+);
 
-  final String title;
-  final String text;
-  final WidgetBuilder builder;
+// Create disabled button
+new FlatButton(
+  child: new Text('BUTTON TITLE')
+);""";
 
-  TabLabel get tabLabel => new TabLabel(text: title.toUpperCase());
+const String _dropdownText =
+  "# Dropdown buttons\n"
+  "A dropdown button selects between multiple selections. The button displays "
+  "the current state and a down arrow. Available states may be shown as a "
+  "list of strings, a palette, or icons, for example.";
 
-  // The TabBarSelection created below saves and restores _ButtonDemo objects
-  // to recover this demo's selected tab. To enable it to compare restored
-  // _ButtonDemo objects with new ones, define hashCode and operator== .
+const String _dropdownCode =
+"""// Member variable holding value
+String dropdownValue
 
-  @override
-  bool operator==(Object other) {
-    if (other.runtimeType != runtimeType)
-      return false;
-    _ButtonDemo typedOther = other;
-    return typedOther.title == title && typedOther.text == text;
-  }
+// Drop down button with string values
+new DropDownButton<String>(
+  value: dropdownValue,
+  onChanged: (String newValue) {
+    setState(() {
+      if (newValue != null)
```

Comment:
A comment here indicating that null means the user dismissed the menu without making a selection would be helpful